### PR TITLE
lex: add page

### PR DESCRIPTION
**Pull request by lmcmenemie**
----

- [x] The page (if new), does not already exist in the repo.

- [x] The page (if new), has been added to the correct platform folder:  
      `common/` if it's common to all platforms, `linux/` if it's Linux-specific, and so on.

- [x] The page has 8 or fewer examples.

- [x] The PR is appropriately titled:  
      `<command name>: add page` for new pages, or `<command name>: <description of changes>` for pages being edited

- [x] The page follows the [contributing](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md) guidelines

### CODE REVIEW

File: pages/common/lex.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+> Lexical analyser generator.
+> Given a lexical analyser, generates C code implementing it.
+
+- Generate analyser from Lex file:
+
+`lex {{path/to/example.l}}`
+
+- Compile a C file generated by lex:
+
+`cc {{path/to/lex.yy.c}} --output {{example}}`
```
Comment (agnivade) [inline pages/common/lex.md]: Should this be `example.l` ?

Comment (agnivade) [reply pages/common/lex.md]: It seems you didn't change it. Was that intended ?

Comment (lmcmenemie) [reply pages/common/lex.md]: This is the same commit we are commenting on. :) Have a look at the latest revision.

Comment (agnivade) [reply pages/common/lex.md]: The latest version of your example says `cc {{path/to/lex.yy.c}} --output {{executable}}`. I believe the idea was to change `example` to `example.l`. I see that you have changed `example` to `executable`. Let me know if I am missing something.

File: pages/common/lex.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# lex
+
+> Lexical analyser generator.
+> Given a lexical analyser, generates C code implementing it.
+
+- Generate analyser from Lex file:
```
Comment (sbrl) [inline pages/common/lex.md]: Perhaps `an analyser` and `a Lex file` would read better here?

Also, you've put `lex` with a capital `L` here, but used a lowercase `l` further down. Is this intentional?

Comment (lmcmenemie) [reply pages/common/lex.md]: Both good points; addressed in new commit. :)

**Comment by agnivade** [comment]
Hey @lmcmenemie, can you take a look at this [comment](https://github.com/tldr-pages/tldr/pull/1783#discussion_r156398869) of mine ? Not sure if you saw it or not.

**Comment by lmcmenemie** [comment]
@agnivade Sorry, saw that but forgot about it. :)

There are at least three files involved when using Lex.

* The Lex file (a description of a lexical analyser)
* The output of Lex (C source code implementing the analyser)
* The compiled source code binary.

The output of the `cc` command is therefore an extensionless binary --- I am not sure what would be the best placeholder name for this. Any ideas?

**Comment by agnivade** [comment]
Right. You had commented that time with "Good point", so I thought maybe that made sense.

**Pull request merged**